### PR TITLE
fix(storage): sync epoch error handling

### LINKED ISSUE #5005: Handle failures in syncing epoch(barrier) to state store
### Is your feature request related to a problem? Please describe.

Currently, we just panic if fail to sync the states of an epoch to state store:
```
pub async fn sync_epoch(&self, epoch: u64) -> (Vec<LocalSstableInfo>, bool) {
      let timer = self
          .core
          .lock()
          .streaming_metrics
          .barrier_sync_latency
          .start_timer();
      let (local_sst_info, sync_succeed) = dispatch_state_store!(self.state_store(), store, {
          match store.sync(epoch).await {
              Ok(sync_result) => (sync_result.uncommitted_ssts, sync_result.sync_succeed),
              // TODO: Handle sync failure by propagating it back to global barrier manager
              Err(e) => panic!(
                  "Failed to sync state store after receiving barrier prev_epoch {:?} due to {}",
                  epoch, e
              ),
          }
      });
      timer.observe_duration();
      (local_sst_info, sync_succeed)
  }
```
It is not an acceptable solution, we should handle the error properly.

### Describe the solution you'd like

Handle the error properly without panic.

### Describe alternatives you've considered

- Retry with bounded times
- If still cannot sync successfully after retry, we should abort the checkpoint process 

### Additional context

related #4951

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
- Config retry attempts for the S3 client
- Pass back Hummock sync failure to Meta instead of just panic

## Checklist

~- [ ] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
#4951
Fixed #5005

### CODE REVIEW

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: src/stream/src/task/stream_manager.rs [hunk 1]
```diff
@@ -228,25 +228,26 @@ impl LocalStreamManager {
         result
     }
 
-    pub async fn sync_epoch(&self, epoch: u64) -> (Vec<LocalSstableInfo>, bool) {
+    pub async fn sync_epoch(&self, epoch: u64) -> Result<(Vec<LocalSstableInfo>, bool)> {
```
Comment (hzxa21) [inline src/stream/src/task/stream_manager.rs:231]: How will upper layer handle this error? Will it trigger the recovery process?

**Review by hzxa21 — APPROVED** [review]

File: src/stream/src/task/stream_manager.rs (hunk 1 above)
Comment (StrikeW) [reply src/stream/src/task/stream_manager.rs:231]: Meta will get the error via the `BarrierComplete` rpc response and will trigger the recovery process.

FYI.
```
    async fn barrier_complete_and_commit(
        &self,
        prev_epoch: u64,
        result: MetaResult<Vec<BarrierCompleteResponse>>,
        state: &mut BarrierManagerState,
        tracker: &mut CreateMviewProgressTracker,
        checkpoint_control: &mut CheckpointControl<S>,
    ) {
        if let Err(err) = result {
            fail_point!("inject_barrier_err_success");
            let fail_node = checkpoint_control.fail();
            tracing::warn!("Failed to commit epoch {}: {:?}", prev_epoch, err);
            self.do_recovery(err, fail_node.into_iter(), state, tracker)
                .await;
            return;
        }
        ...
      }
```

File: src/object_store/src/object/s3.rs [hunk 2]
```diff
@@ -495,8 +496,11 @@ impl S3ObjectStore {
     ///
     /// See [AWS Docs](https://docs.aws.amazon.com/sdk-for-rust/latest/dg/credentials.html) on how to provide credentials and region from env variable. If you are running compute-node on EC2, no configuration is required.
     pub async fn new(bucket: String, metrics: Arc<ObjectStoreMetrics>) -> Self {
-        let shared_config = aws_config::load_from_env().await;
-        let client = Client::new(&shared_config);
```
Comment (MrCroxx) [inline src/object_store/src/object/s3.rs:499]: This change may cause credentials lost.

**Pull request merged**